File: algorithm016/Week_02/algorithm016/hash/TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        m = {}
        i = 0
        for num in nums:
            m[num] = i
            i = i + 1

        j = 0
        for num in nums:
            c = target - num
            if m.has_key(c) and m.get(c) != j:
                return [m.get(c), j]
            j = j + 1

File: algorithm016/Week_02/algorithm016/hash/TwoSum.java
package algorithm016.hash;

import java.util.HashMap;
import java.util.Map;

public class TwoSum {

	public int[] twoSum(int[] nums, int target) {

		Map<Integer, Integer> finds = new HashMap<Integer,Integer>();

		for (int i = 0; i < nums.length; i++) {
			finds.put(nums[i], i);
		}
		
		for (int i = 0; i < nums.length; i++) {
			int com = target - nums[i];
			if(finds.containsKey(com) && finds.get(com) != i) {
				int j = finds.get(com);
				return new int[] {i,j};
			}
		}
		
		return null;

	}

	public int[] twoSum2(int[] nums, int target) {
        
	}
	
	public static void main(String[] args) {

		TwoSum ts = new TwoSum();

		int[] nums = new int[] {3,2,4};

		int[] result = ts.twoSum(nums, 6);
		
		System.out.println("====>>"+ result[0]+result[1]);
		
	}

}
